modules/core/src/utils/histo-pyramid.js

```diff
@@ -0,0 +1,132 @@
+import Transform from '../core/transform';
+import {cloneTextureFrom} from '../webgl-utils/texture-utils';
+import {log} from '../utils';
+import GL from '../constants';
+
+export const HP_BUILD_VS_UTILS = `\
+// returns the top left texture coordiante corresponding to 4X4 block in higher level texture.
+// size: lower level texture size
+// scale: usually (2, 2)
+// offset: offset with-in 4X4 block of higher level texture
+vec2 hp_getTexCoord(vec2 size, vec2 scale, vec2 offset) {
```

Comment:
I'd type out `histoPyramid` here. It is an unusual library/shader module and the name is not too long.